src/component/project/Project/ProjectFeatureToggles/hooks/useEnvironmentsRef.ts

```diff
@@ -0,0 +1,15 @@
+import { useRef } from 'react';
+
+/**
+ * Don't revalidate if array content didn't change.
+ * Needed for `columns` memo optimization.
+ */
+export const useEnvironmentsRef = (environments: string[] = []) => {
```

Comment:
It's a property on data, on SWR response. I did it this way, because it's needed only on this page.

Reply:
Right, since `environments` is a property of the response, shouldn't it already be stable (until the next fetch)?

Seems fine to use it directly in the dependencies array then (if that's what we're after).

Reply:
O, I get it. For this screen we need to make it stable even after refetch, because it is used in the _memoization_ for _columns_ - the structure of the table, that shouldn't be recalculated every time `features` are changing.